source/nanoFramework.CoreLibrary/System/Guid.cs

```diff
@@ -356,5 +372,118 @@ private int GetResult(uint me, uint them)
+        public static bool TryParseGuidWithDashes(String guidString, out Guid result)
+        {
+            int startPos = 0;
+            int temp;
+            long templ;
+            int currentPos = 0;
+            result = new Guid();
+
+            // check to see that it's the proper length
+            if (guidString[0] == '{')
+            {
+                if (guidString.Length != 38 || guidString[37] != '}')
+                    return false;
+                startPos = 1;
+            }
+            else if (guidString.Length != 36)
+                return false;
+
+            if (guidString[8 + startPos] != '-' ||
+                guidString[13 + startPos] != '-' ||
+                guidString[18 + startPos] != '-' ||
+                guidString[23 + startPos] != '-')
+                return false;
+
+            currentPos = startPos;
+            if (!StringToInt(guidString, ref currentPos, 8, out temp))
+                return false;
+            result._a = temp;
+            ++currentPos; //Increment past the '-';
+
+            if (!StringToInt(guidString, ref currentPos, 4, out temp))
+                return false;
+            result._b = (short)temp;
+            ++currentPos; //Increment past the '-';
+
+            if (!StringToInt(guidString, ref currentPos, 4, out temp))
+                return false;
+            result._c = (short)temp;
+            ++currentPos; //Increment past the '-';
+
+            if (!StringToInt(guidString, ref currentPos, 4, out temp))
+                return false;
+            ++currentPos; //Increment past the '-';
+            startPos = currentPos;
+
+            if (!StringToLong(guidString, ref currentPos, 12, out templ))
+                return false;
+
+            result._d = (byte)(temp >> 8);
+            result._e = (byte)(temp);
+            temp = (int)(templ >> 32);
+            result._f = (byte)(temp >> 8);
+            result._g = (byte)(temp);
+            temp = (int)(templ);
+            result._h = (byte)(temp >> 24);
+            result._i = (byte)(temp >> 16);
+            result._j = (byte)(temp >> 8);
+            result._k = (byte)(temp);
+
+            return true;
+        }
+
+        /// <summary>
+        /// Using StringToLong to convert a hex sub-string to an int, while incrementing the parsePos.
+        /// </summary>
+        /// <param name="str">The string containing the hex sub-string.</param>
+        /// <param name="parsePos">The position of the hex sub-string within str.</param>
+        /// <param name="requiredLength">The length of the hex sub-string.</param>
+        /// <param name="result">The numeric value of the hex string.</param>
+        /// <returns>False if any character is not a hex digit or string is shorter than needed for the requiredLength. Otherwise true.</returns>
+        private static bool StringToInt(String str, ref int parsePos, int requiredLength, out int result)
```

Comment:
The CLR has the `Parse(…)` method for numeric types. Any reason for not using it and providing a specific method for this? 🤔 
Besides following the DRY principle we avoid adding size to the assembly, which should always be a top concern when coding for embedded systems.

Reply:
To use the builtin parse it would have to support the form: 
int.TryParse(guidString.Substring(currentPos, 4), System.Globalization.NumberStyles.HexNumber, System.Globalization.CultureInfo.InvariantCulture, out temp))
Which I believe is not in the CLR?

Reply:
I can however use the Convert.ToInt64(substr, 16)

Reply:
That should work!